lib/rules/no-restricted-exports.js

```diff
@@ -63,6 +114,44 @@ module.exports = {
                     messageId: "restrictedNamed",
                     data: { name }
                 });
+                return;
+            }
+
+            if (name === "default") {
+                if (node.parent.type === "ExportAllDeclaration") {
+                    const isSourceSpecified = node.parent.source;
+
+                    if (isSourceSpecified && restrictDefaultExports && restrictDefaultExports.namespaceFrom) {
```

Comment:
```suggestion

                    if (restrictDefaultExports && restrictDefaultExports.namespaceFrom) {
```

I believe checking if the source is specified is redundant because ExportAllDeclaration must have the source.